week1/search.py

```diff
@@ -109,13 +112,73 @@ def query():
 def create_query(user_query, filters, sort="_score", sortDir="desc"):
     print("Query: {} Filters: {} Sort: {}".format(user_query, filters, sort))
     query_obj = {
-        'size': 10,
-        "query": {
-            "match_all": {} # Replace me with a query that both searches and filters
+        "size": 13,
+        "query" : {
+            "function_score" : {
+                        "query": { "bool" : {
+                            "must" : {
+                                "query_string" : {
+                                    "fields" : ["name^100", "shortDescription^50", "longDescription^10", "department"],
+                                    "query" : user_query,
+                                    "phrase_slop": 3
+                                }
+                            },
+                            "filter": filters
+                            }
+                        },
+                        "boost_mode": "multiply",
+                        "score_mode": "avg",
+                        "functions": [
+                            {"field_value_factor": {
+                            "field": "salesRankShortTerm",
+                            "missing": 100000000,
+                            "modifier": "reciprocal"
+                            }},
+                            {"field_value_factor": {
+                            "field": "salesRankMediumTerm",
+                            "missing": 100000000,
+                            "modifier": "reciprocal"
+                            }},
+                            {"field_value_factor": {
+                            "field": "salesRankLongTerm",
+                            "missing": 100000000,
+                            "modifier": "reciprocal"
+                            }}
+                        ]
+            }
         },
         "aggs": {
-            #### Step 4.b.i: create the appropriate query and aggregations here
-
-        }
+            "regularPrice": {
+                "range": {
+                    "field": "regularPrice",
+                    "ranges": [
+                        {"to": 15.0},
```

Comment:
I'm curious on how this looks on the UI. but if you want it to look like the screenshot on the project instructions, you can add a label ("key") to each range. This is how I did it:

```
            "regularPrice": {
                "range":{
                    "field": "regularPrice",
                    "ranges": [
                        { "key": "$", "to": 99},
                        { "key": "$$", "from": 99, "to": 200},
                        { "key": "$$$", "from": 200, "to": 300  },
                        { "key": "$$$$", "from": 300, "to": 400 },
                        { "key": "$$$$$", "from": 400, "to": 500 },
                        { "key": "$$$$$$", "from": 500 }
                    ]
                }
            },
```

Reply:
@lucila Thank you, this makes sense. Mine looked like this 
<img width="268" alt="image" src="https://user-images.githubusercontent.com/36734709/194943396-4ac9ff71-99ed-4f2b-84af-62fd36237ec7.png">